sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -218,14 +219,35 @@ case class CatalogTable(
     tableType: CatalogTableType,
     storage: CatalogStorageFormat,
     schema: Seq[CatalogColumn],
-    partitionColumns: Seq[CatalogColumn] = Seq.empty,
-    sortColumns: Seq[CatalogColumn] = Seq.empty,
+    partitionColumnNames: Seq[String] = Seq.empty,
```

Comment:
The hive syntax has PARTITIONED BY (col_name  data_type  [COMMENT col_comment], ...). The column definition here is the same as the ones in schema. CatalogColumn can represent these 3 aspects of each column.. why we need to change to use String here?